website/static/js/contribAdder.js

```diff
@@ -40,6 +40,25 @@ var AddContributorViewModel = oop.extend(Paginator, {
         self.parentTitle = parentTitle;
         self.async = options.async || false;
         self.callback = options.callback || function() {};
+        self.nodesOriginal = {};
+        //state of current nodes
+        self.nodesToChange = ko.observableArray();
+        self.nodesState = ko.observableArray();
+        //nodesState is passed to nodesSelectTreebeard which can update it and key off needed action.
+        self.nodesState.subscribe(function(newValue) {
+            //The subscribe causes treebeard changes to change which nodes will be affected
+            var nodesToChange = [];
+            for (var key in newValue) {
+                newValue[key].changed = newValue[key].checked !== self.nodesOriginal[key].checked;
+                if (newValue[key].changed) {
+                    nodesToChange.push(key);
+                }
+            }
+            self.nodesToChange(nodesToChange);
+            m.redraw(true);
+        });
+
+        self.hasChildren = ko.observable(false);
```

Comment:
`hasChildren` is derived from `nodesOriginal`, so it should probably be a `computed` rather than an `observable`.